File: 04-hibernate/UTumilovichJPAPractice/src/Supplier.java
import javax.persistence.*;
import java.util.Set;
import java.util.HashSet;

@Entity
public class Supplier extends Company {
    private String bankAccountNumber;

    @OneToMany
    @JoinColumn(name="Supplier_FK")
    private final Set<Product> products = new HashSet<>();

    public Supplier() {
    }

    public Supplier(String companyName, String country, String city, String street,
                    String zipCode, String bankAccountNumber) {
        super(companyName, country, city, street, zipCode);
        this.bankAccountNumber = bankAccountNumber;
    }

    public String getBankAccountNumber() {
        return bankAccountNumber;
    }

    public void setBankAccountNumber(String bankAccountNumber) {
        this.bankAccountNumber = bankAccountNumber;
    }

    public Set<Product> getProducts() {
        return products;
    }

    public void addProduct(Product product) {
        this.products.add(product);
        product.setSupplier(this);
    }
}

File: entity-framework/lTumilovichProductEF/lTumilovichProductEF/Supplier.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lTumilovichProductEF
{
    class Supplier:Company
    {
        public Supplier()
        {
            Products = new List<Product>();
        }

        public string BankAccountNumber { get; set; }
        public List<Product> Products { get; set; }
    }
}
